### PR TITLE
CLOUDP-168386: Re-create Shared CLI Commands as Atlas-only Commands

### PR DESCRIPTION
**Pull request by matt-condon**
## Proposed changes

_Jira ticket:_ CLOUDP-168386

To ease the SDK v2 integration efforts for commands shared by atlas and mongocli, we have split the shared commands such that we now have a set of commands specifically for atlas, and another set for mongocli

### In this PR

Recreated commands for the following:
- Organizations
- Projects
- Teams
- Users
- Alerts
- Performance-Advisor
- Events
- Serverless
- Backup

Additionally:
- Removing mongocli references from atlas versions of commands
- Removing previously created conditional init output opts for multi-template approach
- Updating date in headers for new/renamed files

No v2 sdk integrations are included in this PR

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/CONTRIBUTING.md) (if appropriate)
- [ ] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [ ] I have updated [test/README.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/test/README.md) (if an e2e test has been added)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

**Review by gssbzn — COMMENTED** [review]
I don't see any change to tests, how do you know the new duplicated commands are being tested?

**Comment by wtrocki** [comment]
> I don't see any change to tests, how do you know the new duplicated commands are being tested?

 Did number of smoke tests to verify if documentation/help of the commands weren't affected (e2e tests do not test that).
Also did some manual testing. The whole form of this PR was to make it so clean that we do not need to do much more verification

**Comment by wtrocki** [comment]
Also Printing cobra tree out of the atlas cli before and after this PR does return exactly the same result.

**Comment by matt-condon** [comment]
> I don't see any change to tests, how do you know the new duplicated commands are being tested?

There are no required changes to e2e tests since this PR is just duplicating the atlas commands and updating the references in  [atlas/builder.go](https://github.com/mongodb/mongodb-atlas-cli/blob/6a4bc44c8210b78f10e8d62e01dbb7fad3b00587/internal/cli/root/atlas/builder.go) so the behaviour and as such the e2e test execution should be exactly the same.

Since the Atlas versions of the existing commands were removed/de-referenced in `builder.go`, the Atlas e2e tests wouldn't execute/pass if they weren't being run via these new commands

**Review by fmenezes — APPROVED** [review]
@matt-condon great work here :tada:

I don't think we need extra e2e tests but I've scheduled the Cloud Manager and Ops Manager tests.

Make sure to double-check those tests before merging.

File: internal/cli/root/atlas/builder.go [hunk 1]
```diff
@@ -212,7 +212,7 @@ func Builder() *cobra.Command {
 		maintenance.Builder(),
 		customdns.Builder(),
 		cloudproviders.Builder(),
-		serverless.AtlasCLIBuilder(),
+		serverless.Builder(),
```
Comment (matt-condon) [inline internal/cli/root/atlas/builder.go:215]: Most relevant changes are in this file (commenting for reference)

File: internal/cli/mongocli/serverless/backup/backup.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
+// Copyright 2023 MongoDB Inc
```
Comment (fmenezes) [inline internal/cli/mongocli/serverless/backup/backup.go:1]: I don't think we need this entire folder, double check if you duplicated other files without need.

File: docs/mongocli/command/mongocli-iam-organizations-describe.txt [hunk 3]
```diff
@@ -85,7 +85,7 @@ If the command succeeds, the CLI returns output similar to the following sample.
 .. code-block::
 
    ID     NAME
-   <Id>   <Name>
+   <ID>   <Name>
```
Comment (matt-condon) [inline docs/mongocli/command/mongocli-iam-organizations-describe.txt:88]: This reverts a previous change as the Id format is only used by atlas sdk

[see atlas version here](https://github.com/mongodb/mongodb-atlas-cli/blob/6a4bc44c8210b78f10e8d62e01dbb7fad3b00587/docs/atlascli/command/atlas-organizations-describe.txt#L85-L88)

File: internal/cli/mongocli/serverless/backup/backup.go (hunk 2 above)
Comment (matt-condon) [reply internal/cli/mongocli/serverless/backup/backup.go:1]: A bit easier to review if looking specifically at the [Recreate serverless commands for MongoCLI commit](https://github.com/mongodb/mongodb-atlas-cli/pull/1841/commits/969a5486c09ed3c67698522bb889eb4fe5355c3c#diff-63567b515afffd64b3bbef2fce4778433d3c713843e7fe2e10f11a724d23cd7f), but previously we had a Builder for [Atlas CLI](https://github.com/mongodb/mongodb-atlas-cli/blob/master/internal/cli/root/atlas/builder.go#L215) `AtlasCLIBuilder()` and the builder [for MongoCLI](https://github.com/mongodb/mongodb-atlas-cli/blob/6a4bc44c8210b78f10e8d62e01dbb7fad3b00587/internal/cli/atlas/atlas.go#L87) `Builder()`

TL;DR: `AtlasBuilder()` is used by AtlasCLI and has some additional commands compared to `Builder()`, which is used by MongoCLI, hence the split

Comment (matt-condon) [reply internal/cli/mongocli/serverless/backup/backup.go:1]: Everything that's duplicated previously had two versions of the Builders, one for AtlasCLI and one for MongoCLI, so each of those has been split and the callers updated with the appropriate references

File: internal/cli/root/atlas/builder.go [hunk 4]
```diff
@@ -189,8 +189,8 @@ func Builder() *cobra.Command {
 		auth.Builder(),
 		quickstart.Builder(),
 		setup.Builder(),
-		projects.AtlasCLIBuilder(),
-		organizations.AtlasCLIBuilder(),
+		projects.Builder(),
+		organizations.Builder(),
```
[the remark this answers is not in the document]
Comment (matt-condon) [reply internal/cli/root/atlas/builder.go:193]: Ah I see what you mean.
So currently we only test the `atlasCLI` commands for IAM via `mongoCLI atlas` command e2e tests? 

I'll create a separate story and get on it 🫡

**Comment by matt-condon** [comment]
Merging this one and raising e2e tests as a follow-up

File: internal/cli/root/atlas/builder.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (matt-condon) [reply internal/cli/root/atlas/builder.go:193]: Addressed here https://github.com/mongodb/mongodb-atlas-cli/pull/1853
